component/loki/source/syslog/internal/syslogtarget/transport.go

```diff
@@ -430,16 +431,32 @@ func (t *UDPTransport) handleRcv(c *ConnPipe) {
 	defer t.openConnections.Done()
 
 	lbs := t.connectionLabels(c.addr.String())
-	err := syslogparser.ParseStream(c, func(result *syslog.Result) {
-		if err := result.Error; err != nil {
-			t.handleMessageError(err)
-		} else {
-			t.handleMessage(lbs.Copy(), result.Message)
+
+	for {
```

Comment:
Thanks for the fix! Can you add a comment here so we don't accidentally revert this code? We've been gradually making sure we've synchronized our code with Promtail, so this is at risk of being reverted if we don't add a warning like this.

```suggestion
	// NOTE: The below code purposefully desyncs from Promtail to fix a bug where 
	// UDP messages were never processed. See grafana/agent#5197 for more context.
	for {
```

Reply:
@rfratto I actually made a simultaneous PR over to the loki repo, since after investigating it appeared to be in both places for the same reason. The next sync should bring that change over, given it's merged in (once I add those tests @erikbaranowski requested) https://github.com/grafana/loki/pull/10708 (though appears the build is failing due to an unrelated test, I can investigate once I finish these)

Reply:
Ah, gotcha, thanks!